src/settings.cpp

```diff
@@ -2060,8 +2041,10 @@ bool SetSettingValue(const StringSettingDesc *sd, std::string value, bool force_
 void StringSettingDesc::ChangeValue(const void *object, std::string &newval) const
 {
 	this->Validate(newval);
+	if (this->pre_check != nullptr && !this->pre_check(newval)) return;
```

Comment:
Should the internal `Validate` happen before or after the `pre_check`? I ask as your doxygen comment says `pre_check` can change the value, possibly making it fail `Validate`? Not sure which one is more important :)

Reply:
In my opinion this is the order to use, as I would assume that the **check** for a setting does not make the setting invalid. Similarly the `Validate` can make what has been checked invalid. Take for example the setting "display_opt". That uses a bitmask with values from DisplayOptions which does not have a value for bit 4. So practically the check could just remove bit 4 from the setting, but if the value would be 256 then the pre check would not remove bit 4 but the validation clamps it to 255 which has bit 4 set.

Reply:
I was thinking the other way around: your name doesn't fit, so `Validate` fails, but it is mostly full of whitespace, so `pre_check` would remove that :P I am fine either way, was just making sure there was a reason you went for this order.